### PR TITLE
Add NFT auto pinning feature flag

### PR DESCRIPTION
**Pull request by cypt4**
Resolves https://github.com/brave/brave-browser/issues/28073

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by yrliou — COMMENTED** [review]
Should it be NftPinning instead of NftAutoPinning?
Also, do we consider also exposing the flag via brave://flags to users? It would be easier for people to do testing without using command line.

File: browser/about_flags.cc [hunk 1]
```diff
@@ -310,6 +310,10 @@ constexpr char kBraveSuperReferralName[] = "Enable Brave Super Referral";
 constexpr char kBraveSuperReferralDescription[] =
     "Use custom theme for Brave Super Referral";
 
+constexpr char kBraveWalletNftPinningFeatureName[] = "Enable NFT pinning";
+constexpr char kBraveWalletNftPinningFeatureDescription[] =
+    "Enables NFT pinning entry points for the Brave Wallet";
```
Comment (yrliou) [inline browser/about_flags.cc:315]: nit: Suggest to use `Enable NFT pinning for Brave Wallet`.

File: components/brave_wallet_ui/common/async/__mocks__/bridge.ts [hunk 2]
```diff
@@ -248,7 +248,8 @@ export class MockedWalletApiProxy {
         isFilecoinEnabled: true,
         isWalletBackedUp: true,
         isWalletCreated: true,
-        isWalletLocked: false
+        isWalletLocked: false,
+        isNftAutoPinningFeatureEnabled: false,
```
Comment (yrliou) [inline components/brave_wallet_ui/common/async/__mocks__/bridge.ts:252]: isNftPinningFeatureEnabled, same for other files below.

**Review by yrliou — APPROVED** [review]

**Comment by yrliou** [comment]
Just a friendly reminder, please squash review/build fix commits into the first commit before merging, thanks.

**Review by Douglashdaniel — APPROVED** [review]
++ Front-end!

File: components/brave_wallet_ui/stories/mock-data/mock-wallet-state.ts [hunk 3]
```diff
@@ -330,5 +330,6 @@ export const mockWalletState: WalletState = {
   passwordAttempts: 0,
   isLoadingCoinMarketData: false,
   coinMarketData: mockCoinMarketData,
-  assetAutoDiscoveryCompleted: false
+  assetAutoDiscoveryCompleted: false,
+  isNftAutopinningFeatureEnabled: false
```
Comment (yrliou) [inline components/brave_wallet_ui/stories/mock-data/mock-wallet-state.ts:334]: isNftPinningFeatureEnabled

File: components/brave_wallet_ui/common/slices/wallet.slice.ts [hunk 4]
```diff
@@ -296,6 +297,8 @@ export const createWalletSlice = (initialState: WalletState = defaultState) => {
         state.accounts = accounts
         state.isWalletBackedUp = payload.isWalletBackedUp
         state.selectedAccount = selectedAccount
+        state.isNftAutopinningFeatureEnabled =
+          payload.isNftAutoPinningFeatureEnabled
```
Comment (yrliou) [inline components/brave_wallet_ui/common/slices/wallet.slice.ts:301]: isNftPinningFeatureEnabled

File: components/brave_wallet_ui/common/slices/wallet.slice.ts [hunk 5]
```diff
@@ -129,7 +129,8 @@ const defaultState: WalletState = {
   onRampCurrencies: [] as BraveWallet.OnRampCurrency[],
   selectedCurrency: undefined,
   passwordAttempts: 0,
-  assetAutoDiscoveryCompleted: false
+  assetAutoDiscoveryCompleted: false,
+  isNftAutopinningFeatureEnabled: false
```
Comment (yrliou) [inline components/brave_wallet_ui/common/slices/wallet.slice.ts:133]: isNftPinningFeatureEnabled

File: components/brave_wallet_ui/common/async/__mocks__/bridge.ts [hunk 6]
```diff
@@ -248,7 +248,8 @@ export class MockedWalletApiProxy {
         isFilecoinEnabled: true,
         isWalletBackedUp: true,
         isWalletCreated: true,
-        isWalletLocked: false
+        isWalletLocked: false,
+        isNftAutoPinningFeatureEnabled: false
```
Comment (yrliou) [inline components/brave_wallet_ui/common/async/__mocks__/bridge.ts:252]: isNftPinningFeatureEnabled

File: browser/about_flags.cc (hunk 1 above)
Comment (yrliou) [reply browser/about_flags.cc:315]: cc @cypt4

**Review by Douglashdaniel — APPROVED** [review]
++

File: components/brave_wallet_ui/constants/types.ts [hunk 7]
```diff
@@ -266,6 +266,7 @@ export interface WalletState {
   selectedCurrency: BraveWallet.OnRampCurrency | undefined
   passwordAttempts: number
   assetAutoDiscoveryCompleted: boolean
+  isNftAutopinningFeatureEnabled: boolean
```
Comment (yrliou) [inline components/brave_wallet_ui/constants/types.ts:269]: isNftPinningFeatureEnabled

File: components/brave_wallet_ui/constants/types.ts [hunk 8]
```diff
@@ -349,6 +350,7 @@ export interface WalletInfoBase {
   accountInfos: AccountInfo[]
   isFilecoinEnabled: boolean
   isSolanaEnabled: boolean
+  isNftAutoPinningFeatureEnabled: boolean
```
Comment (yrliou) [inline components/brave_wallet_ui/constants/types.ts]: isNftPinningFeatureEnabled

**Pull request merged**